get_http.go

```diff
@@ -118,12 +120,60 @@ func (g *HttpGetter) GetFile(dst string, u *url.URL) error {
 		g.Client = httpClient
 	}
 
-	resp, err := g.Client.Get(u.String())
+	// check to see whether user has specified a range of bytes to download.
+	// if user has, but the range is invalid, fall back to downloading the
+	// whole file
+	byteRange, rangeErr := getByteRange(u)
+	if rangeErr != nil {
+		// log that we are downloading whole file even though user
+		// requested a byte range.
+		log.Printf("%s; going to disregard range request and download entire file",
+			rangeErr)
+	}
+
+	// First request here (preReq), is going to be used to determine whether we
+	// can make a ranged request. Second one, (req) is used for the actual
+	// download
+	preReq, err := http.NewRequest("HEAD", u.String(), nil)
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest("HEAD", u.String(), nil)
+	if err != nil {
+		return err
+	}
+
+	isPartialDownload := false
+	if rangeErr == nil && byteRange != nil {
+		// We first make a HEAD request so we can check if the server supports
+		// range queries. If the server/URL doesn't support HEAD requests,
+		// we just fall back to GET.
+		resp, err := g.Client.Do(preReq)
```

Comment:
If this `err` comes back nil, we should take care to close the response body to avoid leaking a file descriptor, as we do below on line 174 for the actual GET request. It looks like net/http doesn't do anything special for HEAD specifically, so we are still required to close the body ourselves. You can probably just immediately close it and move on in this case.